tools-v2/pkg/cli/command/curvebs/list/client/client.go

```diff
@@ -0,0 +1,155 @@
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+/*
+ * Project: CurveCli
+ * Created Date: 2022-11-10
+ * Author: Tsonglew
+ */
+
+package client
+
+import (
+	"context"
+	"fmt"
+
+	cmderror "github.com/opencurve/curve/tools-v2/internal/error"
+	cobrautil "github.com/opencurve/curve/tools-v2/internal/utils"
+	basecmd "github.com/opencurve/curve/tools-v2/pkg/cli/command"
+	"github.com/opencurve/curve/tools-v2/pkg/config"
+	"github.com/opencurve/curve/tools-v2/pkg/output"
+	"github.com/opencurve/curve/tools-v2/proto/proto/nameserver2"
+	"github.com/spf13/cobra"
+	"google.golang.org/grpc"
+)
+
+const (
+	clientExample = `$ curve bs list client`
+)
+
+type ListClientRpc struct {
+	Info          *basecmd.Rpc
+	Request       *nameserver2.ListClientRequest
+	curveFSClient nameserver2.CurveFSServiceClient
+}
+
+var _ basecmd.RpcFunc = (*ListClientRpc)(nil) // check interface
+
+type ClientCommand struct {
+	basecmd.FinalCurveCmd
+	Rpc []*ListClientRpc
+}
+
+var _ basecmd.FinalCurveCmdFunc = (*ClientCommand)(nil) // check interface
+
+func (lRpc *ListClientRpc) NewRpcClient(cc grpc.ClientConnInterface) {
+	lRpc.curveFSClient = nameserver2.NewCurveFSServiceClient(cc)
+}
+
+func (lRpc *ListClientRpc) Stub_Func(ctx context.Context) (interface{}, error) {
+	return lRpc.curveFSClient.ListClient(ctx, lRpc.Request)
+}
+
+func NewClientCommand() *cobra.Command {
+	return NewListCLientCommand().Cmd
+}
+
+func NewListCLientCommand() *ClientCommand {
+	lsCmd := &ClientCommand{
+		FinalCurveCmd: basecmd.FinalCurveCmd{
+			Use:     "client",
+			Short:   "list all client information in curvebs",
+			Example: clientExample,
+		},
+	}
+
+	basecmd.NewFinalCurveCli(&lsCmd.FinalCurveCmd, lsCmd)
+	return lsCmd
+}
+
+// AddFlags implements basecmd.FinalCurveCmdFunc
+func (pCmd *ClientCommand) AddFlags() {
+	config.AddBsMdsFlagOption(pCmd.Cmd)
+	config.AddRpcRetryTimesFlag(pCmd.Cmd)
+	config.AddRpcTimeoutFlag(pCmd.Cmd)
+}
+
+// Init implements basecmd.FinalCurveCmdFunc
+func (pCmd *ClientCommand) Init(cmd *cobra.Command, args []string) error {
+	mdsAddrs, err := config.GetBsMdsAddrSlice(pCmd.Cmd)
+	if err.TypeCode() != cmderror.CODE_SUCCESS {
+		return err.ToError()
+	}
+
+	timeout := config.GetFlagDuration(pCmd.Cmd, config.RPCTIMEOUT)
+	retrytimes := config.GetFlagInt32(pCmd.Cmd, config.RPCRETRYTIMES)
+
+	rpc := &ListClientRpc{
+		Request: &nameserver2.ListClientRequest{},
+		Info:    basecmd.NewRpc(mdsAddrs, timeout, retrytimes, "ListClient"),
+	}
+	pCmd.Rpc = append(pCmd.Rpc, rpc)
+
+	header := []string{cobrautil.ROW_ADDR}
```

Comment:
Separate ip and port, is it better to sort by ip? Like this:
```
+------------------+--------+
|       IP         |  PORT  |
+------------------+--------+
| 10.182.2.48      |  9001  |
+                  +--------+
|                  |  9002  |
+------------------+--------+
```

Reply:
updated
<img width="355" alt="image" src="https://user-images.githubusercontent.com/14333766/202159148-3d9fc78f-09f3-4859-9b14-b885d4112bee.png">